components/templates/TerminalChallenge/TerminalChallengeLayout.tsx

```diff
@@ -0,0 +1,94 @@
+'use client'
+
+import { allLessons, Lesson } from 'contentlayer/generated'
+import { useState, useEffect } from 'react'
+import { useMediaQuery } from 'react-responsive'
+import { Tabs } from 'components/ui/Tabs'
+import { Info } from './Info'
+import { Code } from './Code'
+
+/**
+ * @slug {string} for fetching challenge data
+ * @expectedInput {string} answer to the challenge problem
+ * @next {string} link to next part of chapter
+ * @chapterInfo {string} chapter number / info
+ * @challegneInfo {string} challenge number / info
+ * @copyableCodeContent {string} Copyable code content / text
+ */
+export default function TerminalChallengeLayout({ 
+  slug, 
+  expectedInput, 
+  next,
+	chapterInfo,
```

Comment:
The use of tabs for indentation is really messy in this file, could you please clean that up?

Reply:
@vinayaksh42 you can setup your code editor to format on save https://prettier.io/docs/en/editors.html. Maybe we should just setup a pre-commit hook to format the code so people don't even need to think about it.

Reply:
I have changed prettier to format on save. Yes, I think it will be great to have it as a pre-commit hook